Makefile

```diff
@@ -407,30 +407,30 @@ $(OBJDIR)/% : examples/ceed/%.f | $$(@D)/.DIR
 
 $(OBJDIR)/mfem-% : examples/mfem/%.cpp $(libceed) | $$(@D)/.DIR
 	+$(MAKE) -C examples/mfem CEED_DIR=`pwd` \
-	  MFEM_DIR="$(abspath $(MFEM_DIR))" $*
+	  MFEM_DIR="$(abspath $(MFEM_DIR))" CC=$(CC) CXX=$(CXX) $*
 	mv examples/mfem/$* $@
 
 # Note: Multiple Nek files cannot be built in parallel. The '+' here enables
 #       this single Nek bps file to be built in parallel with other examples,
 #       such as when calling `make prove-all -j2`.
 $(OBJDIR)/nek-bps : examples/nek/bps/bps.usr examples/nek/nek-examples.sh $(libceed) | $$(@D)/.DIR
-	+$(MAKE) -C examples MPI=$(MPI) CEED_DIR=`pwd` NEK5K_DIR="$(abspath $(NEK5K_DIR))" nek
+	+$(MAKE) -C examples MPI=$(MPI) FC=$(FC) CEED_DIR=`pwd` NEK5K_DIR="$(abspath $(NEK5K_DIR))" nek
 	mv examples/nek/build/bps $(OBJDIR)/bps
 	cp examples/nek/nek-examples.sh $(OBJDIR)/nek-bps
 
 $(OBJDIR)/petsc-% : examples/petsc/%.c $(libceed) $(ceed.pc) | $$(@D)/.DIR
 	+$(MAKE) -C examples/petsc CEED_DIR=`pwd` \
-	  PETSC_DIR="$(abspath $(PETSC_DIR))" OPT="$(OPT)" $*
+	  PETSC_DIR="$(abspath $(PETSC_DIR))" CC=$(CC) OPT="$(OPT)" $*
```

Comment:
These examples were using the default env CC instead of the one we set. This was a silent problem for GCC/Clang but caused conflicting opt flags vs compilers for ICC.

Reply:
I changed my approach here. I switched to Travis building PETSc with the env CC.